aws-organizations-account/src/main/java/software/amazon/organizations/account/BaseHandlerStd.java

```diff
@@ -183,25 +184,31 @@ public final ProgressEvent<ResourceModel, CallbackContext> handleRetriableExcept
         final AccountConstants.Action actionName,
         final AccountConstants.Handler handlerName
     ) {
-        try {
-            String accountInfo = model.getAccountId() == null ? handlerRequest.getLogicalResourceIdentifier() : model.getAccountId();
-            if (actionName != AccountConstants.Action.CREATE_ACCOUNT) {
-                int currentAttempt = context.getCurrentRetryAttempt(actionName, handlerName);
-                if (currentAttempt < MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION) {
+        String accountInfo = model.getAccountId() == null ? handlerRequest.getLogicalResourceIdentifier() : model.getAccountId();
+
+
+        if (actionName != AccountConstants.Action.CREATE_ACCOUNT) {
+            int currentAttempt = context.getCurrentRetryAttempt(actionName, handlerName);
+            if (currentAttempt < MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION) {
+                context.setCurrentRetryAttempt(actionName, handlerName);
+                if (handlerName == AccountConstants.Handler.READ
+                        || handlerName == AccountConstants.Handler.LIST) {
+                    logger.log(String.format("Got %s when calling %s for "
+                                    + "account [%s]. Retrying %s of %s ",
+                            e.getClass().getName(), organizationsRequest.getClass().getName(), accountInfo, currentAttempt + 1, MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION));
+                    throw RetryableException.builder().cause(e).build();
```

Comment:
Whether CFN will rerun read/list handler logic if it receive this retryableException?

Reply:
When the ProgressEvent framework receive a retryableException, the retry happen within the handler framework, retry is not done by the CFN